src/odbc_connection.cpp

```diff
@@ -674,14 +471,156 @@ Napi::Value ODBCConnection::CreateStatement(const Napi::CallbackInfo& info) {
  ********************************** QUERY *************************************
  *****************************************************************************/
 
+typedef struct QueryOptions {
+  bool         use_cursor;
+  SQLTCHAR    *cursor_name;
+  SQLSMALLINT  cursor_name_length;
+  SQLULEN      fetch_size;
+} QueryOptions;
+
+#define QUERY_OPTION_PROPERTY_NAME_CURSOR     "cursor"
+#define QUERY_OPTION_PROPERTY_NAME_FETCH_SIZE "fetchSize"
+
+QueryOptions
+parse_query_options
+(
+  Napi::Env   env,
+  Napi::Value options_value
+)
+{
+  QueryOptions query_options;
+
+  if (options_value.IsNull())
+  {
+    query_options.use_cursor  = false;
+    query_options.cursor_name = NULL;
+
+    return query_options;
+  }
+
+  Napi::Object options_object = options_value.As<Napi::Object>();
+
+  // .cursor property
+  if (options_object.HasOwnProperty(QUERY_OPTION_PROPERTY_NAME_CURSOR))
+  {
+    Napi::Value cursor_value =
+      options_object.Get(QUERY_OPTION_PROPERTY_NAME_CURSOR);
+    
+    if (cursor_value.IsString())
+    {
+      query_options.use_cursor  = true;
+#ifdef UNICODE
+      std::u16string cursor_string;
+      cursor_string = cursor_value.As<Napi::String>().Utf16Value();
+#else
+      std::string cursor_string;
+      cursor_string = cursor_value.As<Napi::String>().Utf8Value();
+#endif
+      query_options.cursor_name = (SQLTCHAR *)cursor_string.c_str();
+      query_options.cursor_name_length = cursor_string.length();
+    }
+    else if (cursor_value.IsBoolean())
+    {
+      query_options.use_cursor  = cursor_value.As<Napi::Boolean>().Value();
+      query_options.cursor_name = NULL;
+    }
+    else
+    {
+      Napi::TypeError::New(env, "Connection.query options: ." QUERY_OPTION_PROPERTY_NAME_CURSOR " must be a STRING or a BOOLEAN value.").ThrowAsJavaScriptException();
```

Comment:
Of course, making it a variable then you can't do this.

Reply:
Does this make any sense?

```
Napi::TypeError::New(env, std::string("Connection.query options: ." + std::string(QueryOptions::FETCH_SIZE_PROPERTY) + " must be a NUMBER value.")).ThrowAsJavaScriptException();
```

It compiles and works, but just wondering if its too ugly.

Reply:
You can append a `const char*` to a string:

```c++
Napi::TypeError::New(env, std::string("Connection.query options: .") + QueryOptions::FETCH_SIZE_PROPERTY + " must be a NUMBER value.")).ThrowAsJavaScriptException();
```